noxfile.py

```diff
@@ -49,6 +50,43 @@
 nox.options.sessions = ["qa"]
 
 
+def get_gcc_version_to_use():
+    CC = os.environ.get("CC")
+    if CC is None:
+        for gcc in reversed([gcc for gcc in GCC_VERSIONS]):
+            gcc = shutil.which(gcc)
+            if gcc is not None:
+                CC = os.path.split(gcc)[1]
+                break
+        else:
+            CC = shutil.which("gcc")
```

Comment:
I think I've mentioned this previously, that I have a personal preference for early returns. Of course you're not required to rewrite the function, but it could make the data flow clearer:

```python
def get_gcc_verson_to_use() -> str:
    # If the user explicitly set CC variable, use that directly without checks.
    cc = os.environ.get("CC")
    if cc:
        return cc

    # find the first installed compiler version that we support
    for cc in GCC_VERSIONS:
        if shutil.which(cc):
            return cc

    # check if there's a "gcc" or "clang" binary with a suitable version
    for cc in ["gcc", "clang"]:
        ...  # sniff version number from --version message
```

Remarks in that context:

* why reverse the `GCC_VERSIONS` list?
* I think that `gcc` will always equal `os.path.split(shutil.which(gcc))[1]`, unless symlinks are involved. But even in the case of symlinks, we'd rather have the exact name from `GCC_VERSIONS`.

Reply:
Ok, but than we have to split in each return because the user can set the compiler with a full path.
The reverse and the `for cc in ["gcc", "clang"]:` was because the old list was sorted ascending. This can be removed now as I changed the order.